sync-core/src/main/java/com/cloudant/mazha/CouchClient.java

```diff
@@ -180,11 +218,16 @@ public ChangesResult changes(String filterName, Map<String, String> filterParame
         return this.changes(options);
     }
 
-    public ChangesResult changes(Map<String, Object> options) {
-        Preconditions.checkNotNull(options, "options must not be null");
-        URI changesFeedUri = this.uriHelper.changesUri(options);
-        HttpConnection connection = Http.GET(changesFeedUri);
-        return executeToJsonObject(connection, ChangesResult.class);
+    public ChangesResult changes(final Map<String, Object> options) {
+        return executeWithRetry(new Callable<ChangesResult>() {
+            @Override
+            public ChangesResult call() throws Exception {
+                Preconditions.checkNotNull(options, "options must not be null");
```

Comment:
I think we should do the `Preconditions` before calling `executeWithRetry` call.